vnext/Microsoft.ReactNative.Managed/ReactPropertyBag.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Microsoft.ReactNative.Managed
+{
+  public struct ReactPropertyBag
+  {
+    public IReactPropertyBag Handle { get; }
+
+    public bool IsValid => Handle != null;
+
+    public static explicit operator bool(ReactPropertyBag propertyBag) => propertyBag.IsValid;
+
+    public ReactPropertyBag(IReactPropertyBag handle = null)
+    {
+      Handle = handle;
+    }
+
+    // Get property value by property name.
+    public static T Get<T>(IReactPropertyBag handle, ReactPropertyId<T> propertyId)
+    {
+      if (handle == null)
+      {
+        return default(T);
+      }
+      object propertyValue = handle.Get(propertyId.Handle);
+      return (T)propertyValue;
+    }
+
+    // Ensure that property is created by calling valueCreator if needed, and return value by property name.
+    // The TCreateValue must return either T or std::optional<T>. It must have no parameters.
+    public static T GetOrCreate<T>(IReactPropertyBag handle,
+      ReactPropertyId<T> propertyId,
+      Func<T> createValue)
+    {
+      if (handle == null)
+      {
+        return default(T);
+      }
+      object propertyValue = handle.GetOrCreate(propertyId.Handle, () => createValue());
+      return (T)propertyValue;
+    }
+
+    // Set property value by property name.
+    public static void Set<T>(IReactPropertyBag handle, ReactPropertyId<T> propertyId, T value)
+    {
+      if (handle != null)
+      {
+        handle.Set(propertyId.Handle, value);
+      }
+    }
+
+    // Removes property value by property name.
+    public static void Remove<T>(IReactPropertyBag handle, ReactPropertyId<T> propertyId)
+    {
+      if (handle != null)
+      {
+        handle.Set(propertyId.Handle, null);
+      }
+    }
+
+    // Get property value by property name.
```

Comment:
nit: extra newline and not using doccomments.

Reply:
I will add doc comments in a follow up PR.